proto/viam/app/data/v1/data.proto

```diff
@@ -39,6 +39,9 @@ service DataService {
 
   // RemoveTagsToBinaryDataByFilter removes string tags from binary data based on the given filter.
   rpc RemoveTagsFromBinaryDataByFilter(RemoveTagsFromBinaryDataByFilterRequest) returns (RemoveTagsFromBinaryDataByFilterResponse);
+
+  // TagsByFilter gets all tags that match the given filter.
```

Comment:
[nit]: "gets all unique tags" to indicate there will be no duplicates in the returned tags (IIRC)

Reply:
updated!